receiver/prometheusreceiver/internal/otlp_metricfamily.go

```diff
@@ -40,6 +40,7 @@ type metricFamily struct {
 	labelKeysOrdered  []string
 	metadata          *scrape.MetricMetadata
 	groupOrders       map[string]int
+	exemplars         map[uint64]pmetric.ExemplarSlice
```

Comment:
Since all metrics in a metric family have already been determined to have the same base label set, i'm actually not sure we need to do hashing here.  Also, we should be merging all of the exemplars into a single slice in the end anyways...

Let's make this a map of exemplar value to exemplar (not slice): `map[float64]pmetric.Exemplar`.  That way, we de-duplicate on the value, and can just iterate over all exemplars when we construct the data point at the end.